src/app/functions/server/getOneRepResult.ts

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export async function getOneRepResult(
+  method: string,
+  path: string,
+  body?: string
+) {
+  const bearerToken = process.env.ONEREP_API_KEY;
+  const options = {
+    method,
+    headers: {
+      Authorization: `Bearer ${bearerToken}`,
+      "Content-Type": "application/json",
+    },
+  };
+
+  if (body) {
+    // @ts-ignore FIXME
+    options.body = body;
+  }
+
+  const result = await fetch(`${process.env.ONEREP_API_BASE}/${path}`, options);
+  if (!result.ok) {
+    throw new Error(
+      `Error connecting to provider: ${process.env.ONEREP_API_BASE},
+       ${JSON.stringify(await result.json())}`
```

Comment:
Hhmm. is it possible that the `result` may not be JSON object on a `!ok` response and `await result.json()` can throw an error here?
Not entirely sure what that means or how it could happen (since I've never used OneRep API before), but maybe on a 404 path or invalid bearer token/config, or 5xx API server error?

Possibly not a concern since we don't seem to be in a `try..catch` block and presumably the mythical `await result.json()` error might just rethrow upwards to the callee anyways. <i>🤷</i>